tests/unit/src/kakarot/instructions/test_push_operations.cairo

```diff
@@ -16,441 +17,40 @@ from kakarot.instructions.push_operations import PushOperations
-    let result = PushOperations.exec_push14(ctx);
-    TestHelpers.assert_stack_last_element_contains_uint256(
-        result.stack, Uint256(0xFFFFFFFFFFFFFFFFFFFFFFFFFFFF, 0)
-    );
-
-    return ();
-}
-
-@external
-func test__exec_push15_should_add_15_byte_to_stack{
-    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
-}() {
-    alloc_locals;
-    let ctx: model.ExecutionContext* = TestHelpers.init_context_with_bytecode(16, 0xFF);
-    let result = PushOperations.exec_push15(ctx);
-    TestHelpers.assert_stack_last_element_contains_uint256(
-        result.stack, Uint256(0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF, 0)
-    );
-
-    return ();
-}
-
-@external
-func test__exec_push16_should_add_16_byte_to_stack{
-    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
-}() {
-    alloc_locals;
-    let ctx: model.ExecutionContext* = TestHelpers.init_context_with_bytecode(17, 0xFF);
-    let result = PushOperations.exec_push16(ctx);
-    TestHelpers.assert_stack_last_element_contains_uint256(
-        result.stack, Uint256(0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF, 0)
-    );
-
-    return ();
-}
-
-@external
-func test__exec_push17_should_add_17_byte_to_stack{
-    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
-}() {
-    alloc_locals;
-    let ctx: model.ExecutionContext* = TestHelpers.init_context_with_bytecode(18, 0xFF);
-    let result = PushOperations.exec_push17(ctx);
-    TestHelpers.assert_stack_last_element_contains_uint256(
-        result.stack, Uint256(0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF, 0xFF)
-    );
-
-    return ();
-}
-
-@external
-func test__exec_push18_should_add_18_byte_to_stack{
-    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
-}() {
-    alloc_locals;
-    let ctx: model.ExecutionContext* = TestHelpers.init_context_with_bytecode(19, 0xFF);
-    let result = PushOperations.exec_push18(ctx);
-    TestHelpers.assert_stack_last_element_contains_uint256(
-        result.stack, Uint256(0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF, 0xFFFF)
-    );
-
-    return ();
-}
-
-@external
-func test__exec_push19_should_add_19_byte_to_stack{
-    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
-}() {
-    alloc_locals;
-    let ctx: model.ExecutionContext* = TestHelpers.init_context_with_bytecode(20, 0xFF);
-    let result = PushOperations.exec_push19(ctx);
-    TestHelpers.assert_stack_last_element_contains_uint256(
-        result.stack, Uint256(0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF, 0xFFFFFF)
-    );
-
+    let ctx: model.ExecutionContext* = TestHelpers.init_context_with_bytecode(i - 1, 0xFF);
+    PushOperations.exec_push_i(ctx, i);
+    
     return ();
-}
+} 
 
 @external
-func test__exec_push20_should_add_20_byte_to_stack{
+func test__exec_push_should_add_1_through_16_bytes_to_stack{
```

Comment:
naming here is odd. You want to keep it symmetric with the first test, like

```
func test__exec_push_should_raise
func test__exec_push_should_push
```

Reply:
use only one `test__exec_push_should_push` and assert in python, it'll be easier

Reply:
Asserting in python is cleaner for sure, nice tip